Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -791,20 +820,95 @@ public extension ChatChannelController {
             channelModificationFailed(completion)
             return
         }
-
-        guard let messageId = messageId ?? messages.first?.id else {
+        
+        guard let messageId = messageId ?? lastNewestMessageId ?? messages.first?.id else {
             log.error(ClientError.ChannelEmptyMessages().localizedDescription)
             callback { completion?(ClientError.ChannelEmptyMessages()) }
             return
         }
 
-        channelQuery.pagination = MessagesPagination(pageSize: limit, parameter: .greaterThan(messageId))
+        guard !hasLoadedAllNextMessages && !isLoadingNextMessages else {
+            return
+        }
 
+        isLoadingNextMessages = true
+        channelQuery.pagination = MessagesPagination(pageSize: limit, parameter: .greaterThan(messageId))
         updater.update(channelQuery: channelQuery, isInRecoveryMode: false, completion: { result in
-            self.callback { completion?(result.error) }
+            self.isLoadingNextMessages = false
+            switch result {
+            case let .success(payload):
+                self.updateNewestFetchedMessageId(with: payload)
+                self.callback { completion?(nil) }
+            case let .failure(error):
+                self.callback { completion?(error) }
+            }
         })
     }
 
+    /// Load messages around the given message id. Useful to jump to a message which is not loaded yet.
+    ///
+    /// Cleans the current messages of the channel and loads the message with the given id,
+    /// and the messages around it depending on the limit provided.
+    ///
+    /// Ex: If the limit is 25, it will load the message and 12 on top and 12 below it. (25 total)
+    ///
+    /// - Parameters:
+    ///   - messageId: The message id of the message to jump to.
+    ///   - limit: The number of messages to load in total, including the message to jump to.
+    ///   - completion: Callback when the API call is completed.
+    func loadPageAroundMessageId(_ messageId: MessageId, limit: Int? = nil, completion: ((Error?) -> Void)? = nil) {
+        guard let cid = self.cid, isChannelAlreadyCreated else {
+            channelModificationFailed(completion)
+            return
+        }
+
+        guard !isJumpingToMessage else {
+            return
+        }
+
+        isJumpingToMessage = true
+
+        let limit = limit ?? channelQuery.pagination?.pageSize ?? .messagesPageSize
+        channelQuery.pagination = MessagesPagination(pageSize: limit, parameter: .around(messageId))
+        updater.update(
+            channelQuery: channelQuery,
+            isInRecoveryMode: false,
+            onBeforeSavingChannel: { session in
+                session.deleteChannelMessages(cid: cid)
+            },
+            completion: { result in
```

Comment:
Actually not, the way the `Controller <-> Updater` is setup right now, requires that the closure is strongly retained, otherwise it will be automatically deallocated and it won't work. Every other function call in the controller is working this way

Reply:
As we discussed previously in a tech discussion, this is due to testing reasons. And agreed on documenting this internally for now.

Reply:
Did you have a chance to add the documentation for this one somewhere?